transformer_thermal_model/model/thermal_model.py

```diff
@@ -99,13 +108,28 @@ def __init__(
         self.data = temperature_profile
         self.init_top_oil_temp = init_top_oil_temp
 
+        self.check_config()
+
+    def check_config(self) -> None:
+        """Check if the configuration is valid.
+
+        If the transformer is a ThreeWindingTransformer, the input profile must be a ThreeWindingInputProfile.
+        """
```

Comment:
Ik zou de docstring iets simpeler houden. je hoeft niet de if statements in de docstring te benoemen. 
```suggestion
        """Check if the combination of the transformer and input profile the is valid.
        """
```